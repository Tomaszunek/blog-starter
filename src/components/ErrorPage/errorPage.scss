@import 'src/styles/variables.scss';

.errorPage {
    margin:auto;
    width: 50%;
    margin-top: 250px;
    margin-bottom: 250px;
    display: grid;
    grid-template-columns: 1fr 1fr;
    grid-gap: 20px;
    text-align: center;
    color: $font-c;

    h1 {
        font-size: 3em;        
    }

    p {
        font-size: 1.8em;
        margin-left: 15%;
        margin-right: 15%;
        margin-top: 45px;
    }
}