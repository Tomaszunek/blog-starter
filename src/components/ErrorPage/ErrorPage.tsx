import * as React from 'react';
import './errorPage.scss'

export default class ErrorPageComp extends React.Component {    
  public render() {
    return (
      <div className="errorPage">
        <img src="/images/components/404Error.gif" alt=""/>
        <div>
            <h1>Its 404 Page :)</h1>
            <p>Something went wrong if you came here.</p>
            <p>Please if you like, talk to us about this problem.</p>
        </div>
      </div>
    );
  }
}


