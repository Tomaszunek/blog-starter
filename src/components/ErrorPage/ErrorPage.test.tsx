import * as React from 'react';
import { mount } from 'enzyme';
import * as renderer from 'react-test-renderer';
import ErrorPage from './ErrorPage'


describe('Error Page', () => {
    describe('Component should render right', () => {
        let app:any;
        beforeEach(() => {
            app = mount(<ErrorPage/>);
        })
        it('render correctly snapshot component', () => {
            const tree = renderer
            .create(<ErrorPage/>)
            .toJSON();
            expect(tree).toMatchSnapshot();
        })  
        it('Component should render right', () => {      
            expect(app.find('h1')).toHaveLength(1);
            expect(app.find('h1').text()).toEqual("Its 404 Page :)")
            expect(app.find('p')).toHaveLength(2);
            expect(app.find('img')).toHaveLength(1);
        });
    });
  });