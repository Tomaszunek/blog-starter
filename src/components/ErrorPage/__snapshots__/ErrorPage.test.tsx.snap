// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Error Page Component should render right render correctly snapshot component 1`] = `
<div
  className="errorPage"
>
  <img
    alt=""
    src="/images/components/404Error.gif"
  />
  <div>
    <h1>
      Its 404 Page :)
    </h1>
    <p>
      Something went wrong if you came here.
    </p>
    <p>
      Please if you like, talk to us about this problem.
    </p>
  </div>
</div>
`;
